routers/private/serv.go

```diff
@@ -111,8 +111,17 @@ func ServCommand(ctx *context.PrivateContext) {
 
 	owner, err := user_model.GetUserByName(results.OwnerName)
 	if err != nil {
+		if user_model.IsErrUserNotExist(err) {
+			// User is fetching/cloning a non-existent repository
+			log.Warn("Failed authentication attempt (cannot find repository: %s/%s) from %s", results.OwnerName, results.RepoName, ctx.RemoteAddr())
+			ctx.JSON(http.StatusNotFound, private.ErrServCommand{
+				Results: results,
+				Err:     fmt.Sprintf("Cannot find repository: %s/%s", results.OwnerName, results.RepoName),
+			})
+			return
+		}
 		log.Error("Unable to get repository owner: %s/%s Error: %v", results.OwnerName, results.RepoName, err)
-		ctx.JSON(http.StatusInternalServerError, private.ErrServCommand{
+		ctx.JSON(http.StatusForbidden, private.ErrServCommand{
```

Comment:
403 means the repository is exist but user has no access and 404 means the repository is not exist? That will leak something.

Reply:
The old code (below) is 

```
	if !owner.IsOrganization() && !owner.IsActive {
		ctx.JSON(http.StatusForbidden, private.ErrServCommand{
			Results: results,
			Err:     "Repository cannot be accessed, you could retry it later",
		})
		return
	}
```

It was 403 vs 500 before, not too much different. To prevent from leaking, we should re-design the response code and message.

Reply:
This is an internal URL endpoint. The status code is not shown to the user. Look at what cmd/serv.go does with this result instead of looking at what it is returning.